devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -365,9 +366,12 @@ def zwaveEvent(thermostatoperatingstatev1.ThermostatOperatingStateReport cmd) {
 	if (operatingState) {
 		map.name = "thermostatOperatingState"
 		map.value = operatingState
-
-		if (settings.heatdetails == "No") {
-			map.displayed = false
+		map.displayed = false
+		// If the user want to see each of the Idle and Heating events in the event history,
+		// Otherwise don't show them more frequently than 5 minutes.
+		if (settings.heatdetails == "Yes" ||
+				!secondsPast(device.currentState("thermostatOperatingState")?.getLastUpdated(), 60 * 5)) {
```

Comment:
I don't think `secondsPast` is defined.

Reply:
Yeah, it isn't in this DTH. It is in the Zigbee and Maestro ones down at the bottom of the file. @PKacprowiczS You'll need to copy `secondsPast` over.